pkg/helm/watcher/controller/helm_watcher.go

```diff
@@ -125,13 +126,17 @@ func (r *HelmWatcherReconciler) reconcileDelete(ctx context.Context, repository
 		return ctrl.Result{}, err
 	}
 
+	log.Info("deleted repository cache", "namespace", repository.Namespace, "name", repository.Name)
 	// Remove our finalizer from the list and update it
 	controllerutil.RemoveFinalizer(&repository, watcherFinalizer)
 
 	if err := r.Update(ctx, &repository); err != nil {
-		return ctrl.Result{}, err
+		log.Error(err, "failed to update repository to remove the finalizer, re-queuing after 5 seconds", "namespace", repository.Namespace, "name", repository.Name)
+		// note that we don't return the error if we want the 5 second requeue to work. with the error the Result is ignored.
+		return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
```

Comment:
Don't do this...return an error and it will retry and automatically backoff.

Reply:
Huh nice, that's true. It's using a rate limiter by default. Neat.